inst/extdata/packages.R

```diff
@@ -0,0 +1,94 @@
+# Install repgen, production instance, prerequisite packages. Should be safe to
+# run repeatedly without additional intervention.
+
+# devtools is currently only required on a production instance to install gsplot
+# from GitHub (see below). There are plans to "promote" it to CRAN (or GRAN?),
+# but that hasn't happened yet. See 
+# https://usgs-cida.slack.com/archives/aqcu/p1484934743005621 if it still 
+# exists.
+pkgs <- c(
+  "devtools",
+  "dplyr",
+  "evaluate",
+  "htmlTable",
+  "jsonlite",
+  "lazyeval",
+  "knitr",
+  "lubridate",
+  "rmarkdown",
+  "yaml"
+)
+
+# Known production tiers, listed at
+# https://docs.google.com/document/d/1vBOTUPtIdeCTGe7Or2cQGMMOFS_hD545yZ1E6YBGizs/edit
+tiers <-
+  c(
+    "nwissddvasaqcu.cr.usgs.gov",    # DEV
+    "intcida-test.er.usgs.gov",      # QA
+    "nwissdtrasnwisra1.cr.usgs.gov", # Train 1
+    "nwissdtrasnwisra2.cr.usgs.gov", # Train 2
+    "nwisdata.usgs.gov",             # PROD
+    "reporting.nwis.usgs.gov"        # CHS “prod”
+  )
+
+nodename <- Sys.info()["nodename"]
+
+# if this is a production tier...
+if (nodename %in% tiers) {
+  source("installPackages.R")
+} else {
+  # presume it's a development machine with the source checked out
+  source(paste0(getwd(), "/inst/extdata/installPackages.R"))
+}
+
+# all packages except devtools and its prerequisites are held back to older
+# versions, keyed by their repo URLs below
+installPackages(pkgs, "http://mran.microsoft.com/snapshot/2016-03-31")
```

Comment:
I've stared at the diffs between DEV and my workstation quite a lot now. I suspect R's `install.packages()` is smarter now than when the original package installer was written.

Also it's crossed my mind that since we're in the process of doing this rather major re-factor, it might make sense to bring the package snapshot set up to the present date. I've noticed that the package dependency tree for the development packages at least, seems to have changed since then. I would expect the same for the production packages too.